src/main/java/edu/kis/powp/jobs2d/drivers/adapter/monitor/DeviceMonitorDriver.java

```diff
@@ -0,0 +1,36 @@
+package edu.kis.powp.jobs2d.drivers.adapter.monitor;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.drivers.adapter.DriverDecorator;
+import edu.kis.powp.jobs2d.drivers.singleton.DeviceMonitor;
+
+public class DeviceMonitorDriver implements DriverDecorator {
+
+    private Job2dDriver driver;
+
+    public DeviceMonitorDriver() { }
+
+    public DeviceMonitorDriver(Job2dDriver driver) {
+        this.driver = driver;
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        driver.setPosition(x, y);
+        DeviceMonitor.getInstance().calculateSetPositionDistance(x, y);
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        driver.operateTo(x, y);
+        DeviceMonitor.getInstance().calculateOperateToDistance(x, y);
```

Comment:
It would be much better to allow deviceMonitor instance to be injected in the constructor. Now this is fixed.